### PR TITLE
Add Security information to Security Analytics documentation

### LINKED ISSUE #3183: [DOC]Create an OpenSearch Security compatibility topic for Security Analytics
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.**
Most if not all documentation sections for plugins contain a topic that includes information about using the plugin with OpenSearch Security. Security Analytics needs this topic installed in its documentation, too. The information needs to be tailored to the specific requirements of Security Analytics, with attention to permissions.
 

**What other resources are available?**
And example of the needed content from the Alerting plugin:
https://opensearch.org/docs/latest/observing-your-data/alerting/security/

Security Analytics issue for implementation of Security for Security Analytics
[Implement Security for Security Analytics actions](https://github.com/opensearch-project/security-analytics/issues/50)

### PR DESCRIPTION
**Pull request by cwillum**
### Description
Most plugins have a topic in their documentation that provides information about integration with the Security plugin. This adds an equivalent topic for Security Analytics

### Issues Resolved
Created documentation that discusses basic permissions and backend roles for for Security Analytics
Fixes #3183 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cwillum** [comment]
@opensearch-project/security Team, to begin organizing, defining, and rounding out permissions in the documentation, I took the opportunity to categorize cluster permissions and provide references to APIs while adding new permissions for Security Analytics. There are two cluster permissions, however, that I'm not clear on:
1. cluster:admin/reindex/rethrottle
2. cluster:admin/reroute

Do you know what these allow users to do and what API you would associate them with? Thanks.

File: _security/access-control/permissions.md
Comment (peternied) [inline _security/access-control/permissions.md:1]: Great update, really improves the readability.

Would you want to pull in all the permissions updates from https://github.com/opensearch-project/security-dashboards-plugin/pull/1333 into this change or create another issue to follow up?

Comment (cwillum) [reply _security/access-control/permissions.md:1]: @peternied Good to hear, and thanks for the affirmation. 
RE: Would you want to pull in all the permissions updates...
Definitely. But as you guessed, it's probably better to work off a dedicated issue for this. Can probably break ground using this issue, [#2359](https://github.com/opensearch-project/documentation-website/issues/2359). Looking forward to rounding out this section of docs.

**Comment by cwillum** [comment]
@sbcd90 @amsiglan Could one of you two have a look at the new security section I added for Security Analytics? It's borrowed from other plugins (e.g., alerting, anomaly detection) and includes information like basic roles for the plugin and how to use backend roles. It also cross references to the Permissions page in documentation. This now lists the permissions for Security Analytics. 
I also began reorganizing all cluster permissions on this page. It probably wasn't a good idea to mix the two tasks. But I couldn't resist the change. You'll see a lot of that in the changes, too. Thanks.

File: _security/access-control/permissions.md [hunk 1]
```diff
@@ -134,37 +172,117 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
+
+### Security Analytics permissions
+
+See [API tools]({{site.url}}{{site.baseurl}}/security-analytics/api-tools/index/).
+
+**Permission** | **Description**
+:--- | :---
+cluster:admin/opensearch/securityanalytics/alerts/get | permission to get alerts
+cluster:admin/opensearch/securityanalytics/alerts/ack | permission to acknowledge alerts
+cluster:admin/opensearch/securityanalytics/detector/get | permission to get detectors
```
Comment (amsiglan) [inline _security/access-control/permissions.md:265]: @sbcd90 Correct me if I am wrong but I think `get` will only get one detector with given Id so we should maybe say `permission to get a detector`

**Comment by amsiglan** [comment]
> @sbcd90 @amsiglan Could one of you two have a look at the new security section I added for Security Analytics? It's borrowed from other plugins (e.g., alerting, anomaly detection) and includes information like basic roles for the plugin and how to use backend roles. It also cross references to the Permissions page in documentation. This now lists the permissions for Security Analytics. I also began reorganizing all cluster permissions on this page. It probably wasn't a good idea to mix the two tasks. But I couldn't resist the change. You'll see a lot of that in the changes, too. Thanks.

Left a couple of comments, rest looks good to me

File: _security-analytics/security.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  "password": "bob",
+  "backend_roles": [
+    "human-resources"
+  ],
+  "attributes": {}
+}
+```
+{% include copy-curl.html %}
+
+Both `alice` and `bob` have full access to Security Analytics:
```
[the remark this answers is not in the document]
Comment (cwillum) [reply _security-analytics/security.md:74]: @amsiglan Agree completely. Good point. I've reworded this and the next sentence to clarify what's going on.

File: _security/access-control/permissions.md (hunk 1 above)
Comment (cwillum) [reply _security/access-control/permissions.md:265]: @amsiglan @sbcd90 This is also a valid point. However, these descriptions of the permissions are just stating generally what they're used for. Kind of like, "use this lug wrench to remove the lug bolts." It will remove all of them. But you can only use the wrench on one at a time. And it's consistent with the other descriptions. With the link to the APIs page above the table, you should be able to go directly to the API to learn how it's used.
https://opensearch.org/docs/latest/security-analytics/api-tools/detector-api/#get-detector-api

But to that point, here's the description for the Get Detector API: "The Get Detector API retrieves the detector details. Use the detector ID in the call to fetch detector details."
Do you think I should add that you can only use one `detector_id` at a time? Or do you think it's understood by the description and examples?

File: _security-analytics/security.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    "bob"
+  ]
+}
+```
+{% include copy-curl.html %}
+
+However, because they have different backend roles, `alice` and `bob` cannot view each other's detectors or their results.
+
+
+## A note on using fine-grained access control with the plugin
```
Comment (vagimeli) [inline _security-analytics/security.md:93]: Can we just say "Using...?"

File: _security/access-control/permissions.md [hunk 4]
```diff
@@ -69,11 +69,24 @@ Rather than individual permissions, you can often achieve your desired security
 
 These permissions are for the cluster and can't be applied granularly. For example, you either have permissions to take snapshots (`cluster:admin/snapshot/create`) or you don't. You can't have permissions to take snapshots only for certain indices.
```
Comment (vagimeli) [inline _security/access-control/permissions.md:70]: Suggestion: You can't have partial permissions. For example, you can't have permission to take snapshots of some indexes only.

File: _security/access-control/permissions.md [hunk 5]
```diff
@@ -134,37 +172,117 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
 - cluster:admin/snapshot/create
 - cluster:admin/snapshot/delete
 - cluster:admin/snapshot/get
 - cluster:admin/snapshot/restore
 - cluster:admin/snapshot/status
 - cluster:admin/snapshot/status*
+
+### Task permissions
+
+See [Tasks]({{site.url}}{{site.baseurl}}/api-reference/tasks/) APIs.
```
Comment (vagimeli) [inline _security/access-control/permissions.md:251]: Should APIs be part of the hyperlinked text, like preceding sections?

File: _security/access-control/permissions.md [hunk 6]
```diff
@@ -134,37 +172,117 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
+cluster:admin/opensearch/securityanalytics/mapping/update | permission to update field mappings
+cluster:admin/opensearch/securityanalytics/rules/categories | permission to get all rule categories
+cluster:admin/opensearch/securityanalytics/rule/write | permission to create and update rules
+cluster:admin/opensearch/securityanalytics/rule/search | permission to search for rules
+cluster:admin/opensearch/securityanalytics/rules/validate | permission to validate rules
+cluster:admin/opensearch/securityanalytics/rule/delete | permission to delete rules
+
+### Cluster monitoring
+
+The monitor cluster permissions apply to read-only operations such as checking cluster health and getting information about usage on nodes or tasks executing in the cluster.
```
Comment (vagimeli) [inline _security/access-control/permissions.md:282]: add comma after operations

File: _security/access-control/permissions.md [hunk 7]
```diff
@@ -180,7 +298,11 @@ These permissions are for the cluster and can't be applied granularly. For examp
 - cluster:monitor/task/get
 - cluster:monitor/tasks/list
 
-The following permissions are for indexes but apply globally to the cluster:
```
Comment (vagimeli) [inline _security/access-control/permissions.md:183]: ... indexes, but they apply...

File: _security/access-control/permissions.md [hunk 8]
```diff
@@ -180,7 +298,11 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
+### Index templates
+
+The index template permissions are for indexes but apply globally to the cluster.
```
Comment (vagimeli) [inline _security/access-control/permissions.md:303]: ... indexes, but they apply...

File: _security/access-control/users-roles.md [hunk 9]
```diff
@@ -137,7 +141,6 @@ For more detailed summaries of the permissions for each role, reference their ac
 
 The following examples show how you might set up a read-only and a bulk access role.
```
Comment (vagimeli) [inline _security/access-control/users-roles.md:142]: Suggestion: ...a read-only role and a bulk access role. OR ...read-only and bulk access roles.

**Review by vagimeli — APPROVED** [review]
@cwillum Review completed. Minimal comments.

File: _security-analytics/security.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+You can use OpenSearch Security with Security Analytics to assign user permissions and manage the actions that users can and cannot perform. For example, you might want one group of users to be able to create, update, or delete detectors and another group of users to only view detectors.
+
+All Security Analytics indexes are protected as system indexes. Only a super admin user or an admin user with a TLS certificate can access system indexes. For more information, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/).
+
+Security for Security Analytics works much the same as [Alerting security]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/security/).
+
+
+## Basic permissions
+
+As an admin user, you can use Security to assign specific permissions to users based on the specific APIs they need to access. For a list of supported APIs, see [API tools]({{site.url}}{{site.baseurl}}/security-analytics/api-tools/index/).
```
[the remark this answers is not in the document]
Comment (cwillum) [reply _security-analytics/security.md:19]: Super admin user (= user with an admin certificate) is a little different than the admin user being referred to here.

File: _security-analytics/security.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+OpenSearch Security has two built-in roles that cover most Security Analytics use cases: `security_analytics_full_access` and `security_analytics_read_access`. A third built-in role allows users to view and acknowledge alerts: `security_analytics_ack_alerts`. For descriptions of each, see [Predefined roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
```
[the remark this answers is not in the document]
Comment (cwillum) [reply _security-analytics/security.md:21]: Right, could use some streamlining. Thanks.
OpenSearch Security has three built-in roles that cover most use cases for Security Analytics: `security_analytics_full_access`, `security_analytics_read_access`, and `security_analytics_ack_alerts`. For descriptions of these and other roles, see [Predefined roles].

File: _security-analytics/security.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+If these roles don't meet your needs, mix and match individual Security Analytics [permissions]({{site.url}}{{site.baseurl}}/security/access-control/permissions/#security-analytics-permissions) to suit your use case. Each action corresponds to an operation in the REST API. For example, the `cluster:admin/opensearch/securityanalytics/detector/delete` permission lets you delete detectors.
+
+
+## (Advanced) Limit access by backend role
+
+Use backend roles to configure fine-grained access to individual detectors based on roles. For example, users of different departments in an organization can view detectors owned by their own department.
```
[the remark this answers is not in the document]
Comment (cwillum) [reply _security-analytics/security.md:28]: Agreed. That's rough as is. I tried this:
For example, backend roles can be assigned to users working in different departments of an organization so that they can view only those detectors owned by the departments in which they work.

File: _security-analytics/security.md (hunk 3 above)
Comment (cwillum) [reply _security-analytics/security.md:93]: Using fine-grained access control with any plugin is perfectly acceptable. This note was added for a special reason, so I think it's better to leave as is so it attracts attention differently than a standard heading.

File: _security/access-control/permissions.md (hunk 4 above)
Comment (cwillum) [reply _security/access-control/permissions.md:70]: Agreed, reworded.

File: _security/access-control/permissions.md (hunk 5 above)
Comment (cwillum) [reply _security/access-control/permissions.md:251]: Just highlighting the text from the level 1 heading of the topic.

File: _security-analytics/security.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+title: OpenSearch Security for Security Analytics
+nav_order: 2
+has_children: false
+---
+
+# OpenSearch Security for Security Analytics
+
+You can use OpenSearch Security with Security Analytics to assign user permissions and manage the actions that users can and cannot perform. For example, you might want one group of users to be able to create, update, or delete detectors and another group of users to only view detectors.
+
+All Security Analytics indexes are protected as system indexes. Only a super admin user or an admin user with a TLS certificate can access system indexes. For more information, see [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/).
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:12]: Is a super admin a different concept from the user with an admin certificate? I thought you had to have the admin cert to modify system indexes. If that's not the case can you define what a "super admin user" is, or link to a place in the docs that defines it?

File: _security-analytics/security.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+Security for Security Analytics works similarly to [Alerting security]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/security/).
+
+
+## Basic permissions
+
+As an admin user, you can use Security to assign specific permissions to users based on the specific APIs they need to access. For a list of supported APIs, see [API tools]({{site.url}}{{site.baseurl}}/security-analytics/api-tools/index/).
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:19]: What is an admin user? Is there a specific backend role this is tied to?

File: _security-analytics/security.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+
+OpenSearch Security has three built-in roles that cover most use cases for Security Analytics: `security_analytics_full_access`, `security_analytics_read_access`, and `security_analytics_ack_alerts`. For descriptions of these and other roles, see [Predefined roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles#predefined-roles).
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:21]: Could this be broken out into a bulleted list? Also, would it be possible to include the descriptions of those three roles in the bullet point list? The next paragraph starts with "If these roles don't meet your needs..." and the journey from the beginning of Line 21 to the beginning of Line 23 for a user is going to necessarily include visiting another page without those roles being defined here (or at least a little commentary).

File: _security-analytics/security.md [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+    "plugins.security_analytics.filter_by_backend_roles": "true"
+  }
+}
+```
+{% include copy-curl.html %}
+
+Now when users view Security Analytics resources in OpenSearch Dashboards (or make REST API calls), they only see detectors created by users who share at least one backend role.
+For example, consider two users: `alice` and `bob`.
+
+`alice` has an analyst backend role:
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:47]: There's a disconnect between this phrase and the following curl template - the sentence says "alice has an analyst backend role" but the subsequent command is assigning the backend role to alice, so the ordering doesn't feel right.

Could this code example be introduced as:

```
Assign `alice` the analyst backend role:
```

Or maybe something like:

```
In the following example, user `alice` is being assigned the `analyst` backend role:
```

File: _security-analytics/security.md [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+  "password": "alice",
+  "backend_roles": [
+    "analyst"
+  ],
+  "attributes": {}
+}
+```
+{% include copy-curl.html %}
+
+`bob` has a human-resources backend role:
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:61]: Same as previous comment - you declare that `bob` has a `human-resources` backend role, but it hasn't been defined yet (the command that follows is setting the backend role).

The phrasing used here would be appropriate if the role was already defined and you were listing it with a GET method.

File: _security-analytics/security.md [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+  "password": "bob",
+  "backend_roles": [
+    "human-resources"
+  ],
+  "attributes": {}
+}
+```
+{% include copy-curl.html %}
+
+Both `alice` and `bob` are assigned the role that gives them full access to Security Analytics:
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:75]: Same issue with ordering - they aren't assigned until after the following API call.

File: _security-analytics/security.md [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+
+To reduce the chances of unintended users viewing metadata that could describe an index, we recommend that administrators enable role-based access control and keep these kinds of design elements in mind when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:97]: Avoid the idiomatic phrase "keep in mind" - maybe rephrase to:

```
To reduce the chance of an unintended user viewing metadata that could describe an index, we recommend that administrators enable role-based access control and consider these design elements when assigning permissions to the intended group of users. See [Limit access by backend role](#advanced-limit-access-by-backend-role) for details.
```

File: _security/access-control/permissions.md [hunk 20]
```diff
@@ -67,13 +67,26 @@ Rather than individual permissions, you can often achieve your desired security
...
+
+See [Ingest APIs]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/).
 
 - cluster:admin/ingest/pipeline/delete
 - cluster:admin/ingest/pipeline/get
 - cluster:admin/ingest/pipeline/put
 - cluster:admin/ingest/pipeline/simulate
 - cluster:admin/ingest/processor/grok/get
+
+### Anamoly Detection permissions
```
Comment (JeffH-AWS) [inline _security/access-control/permissions.md:86]: Spelling error - Anomaly

File: _security/access-control/permissions.md [hunk 21]
```diff
@@ -87,6 +100,11 @@ These permissions are for the cluster and can't be applied granularly. For examp
 - cluster:admin/opendistro/ad/result/search
 - cluster:admin/opendistro/ad/result/topAnomalies
 - cluster:admin/opendistro/ad/tasks/search
+
+### Alerting permissions
+
+See [Alerting API]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/api/).
+
 - cluster:admin/opendistro/alerting/alerts/ack (acknowledge)
```
Comment (JeffH-AWS) [inline _security/access-control/permissions.md:108]: Is `(acknowledge)` part of the permission? If not it should be removed from the line.

File: _security-analytics/security.md [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+## (Advanced) Limit access by backend role
+
+Use backend roles to configure fine-grained access to individual detectors based on roles. For example, backend roles can be assigned to users working in different departments of an organization so that they can view only those detectors owned by the departments in which they work.
+
+First, make sure your users have the appropriate [backend roles]({{site.url}}{{site.baseurl}}/security/access-control/index/). Backend roles usually come from an [LDAP server]({{site.url}}{{site.baseurl}}/security/configuration/ldap/) or [SAML provider]({{site.url}}{{site.baseurl}}/security/configuration/saml/). However, if you use the internal user database, you can use the REST API to [add them manually]({{site.url}}{{site.baseurl}}/security/access-control/api#create-user).
+
+Next, enable the following setting:
+
+```json
+PUT _cluster/settings
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:35]: The API should have a leading slash. i.e. PUT /_cluster/settings

File: _security-analytics/security.md [hunk 23]
```diff
@@ -0,0 +1,97 @@
...
+```
+{% include copy-curl.html %}
+
+Now when users view Security Analytics resources in OpenSearch Dashboards (or make REST API calls), they only see detectors created by users who share at least one backend role.
+For example, consider two users: `alice` and `bob`.
+
+`alice` has an analyst backend role:
+
+```json
+PUT _plugins/_security/api/internalusers/alice
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:50]: The API should have a leading slash. i.e. PUT /_plugins/_security/api/internalusers/alice

File: _security-analytics/security.md [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+  ],
+  "attributes": {}
+}
+```
+{% include copy-curl.html %}
+
+`bob` has a human-resources backend role:
+
+```json
+PUT _plugins/_security/api/internalusers/bob
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:64]: Need leading slash on API path:

PUT /_plugins/_security/api/internalusers/bob

File: _security-analytics/security.md [hunk 25]
```diff
@@ -0,0 +1,97 @@
...
+  ],
+  "attributes": {}
+}
+```
+{% include copy-curl.html %}
+
+Both `alice` and `bob` are assigned the role that gives them full access to Security Analytics:
+
+```json
+PUT _plugins/_security/api/rolesmapping/security_analytics_full_access
```
Comment (JeffH-AWS) [inline _security-analytics/security.md:78]: PUT /_plugins/_security/api/rolesmapping/security_analytics_full_access

File: _security/access-control/permissions.md [hunk 26]
```diff
@@ -134,37 +172,117 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
+cluster:admin/opensearch/securityanalytics/findings/get | permission to get findings
+cluster:admin/opensearch/securityanalytics/mapping/get | permission to get field mappings by index
+cluster:admin/opensearch/securityanalytics/mapping/view/get | permission to get field mappings by index and view mapped and unmapped fields
+cluster:admin/opensearch/securityanalytics/mapping/create | permission to create field mappings
+cluster:admin/opensearch/securityanalytics/mapping/update | permission to update field mappings
+cluster:admin/opensearch/securityanalytics/rules/categories | permission to get all rule categories
+cluster:admin/opensearch/securityanalytics/rule/write | permission to create and update rules
+cluster:admin/opensearch/securityanalytics/rule/search | permission to search for rules
+cluster:admin/opensearch/securityanalytics/rules/validate | permission to validate rules
+cluster:admin/opensearch/securityanalytics/rule/delete | permission to delete rules
```
Comment (JeffH-AWS) [inline _security/access-control/permissions.md:278]: Need to add pipe "|" characters to the beginning and end of each line for compatibility (see https://www.markdownguide.org/extended-syntax/#tables for guidance). It technically renders fine this way, but if we change the underlying markdown interpreter then this table could break. Copy-paste this if you want:

```
| **Permission** | **Description** |
| :--- | :--- |
| cluster:admin/opensearch/securityanalytics/alerts/get | permission to get alerts |
| cluster:admin/opensearch/securityanalytics/alerts/ack | permission to acknowledge alerts |
| cluster:admin/opensearch/securityanalytics/detector/get | permission to get detectors |
| cluster:admin/opensearch/securityanalytics/detector/search | permission to search detectors |
| cluster:admin/opensearch/securityanalytics/detector/write | permission to create and update detectors |
| cluster:admin/opensearch/securityanalytics/detector/delete | permission to delete detectors |
| cluster:admin/opensearch/securityanalytics/findings/get | permission to get findings |
| cluster:admin/opensearch/securityanalytics/mapping/get | permission to get field mappings by index |
| cluster:admin/opensearch/securityanalytics/mapping/view/get | permission to get field mappings by index and view mapped and unmapped fields |
| cluster:admin/opensearch/securityanalytics/mapping/create | permission to create field mappings |
| cluster:admin/opensearch/securityanalytics/mapping/update | permission to update field mappings |
| cluster:admin/opensearch/securityanalytics/rules/categories | permission to get all rule categories |
| cluster:admin/opensearch/securityanalytics/rule/write | permission to create and update rules |
| cluster:admin/opensearch/securityanalytics/rule/search | permission to search for rules |
| cluster:admin/opensearch/securityanalytics/rules/validate | permission to validate rules |
| cluster:admin/opensearch/securityanalytics/rule/delete | permission to delete rules |
```

File: _security/access-control/permissions.md [hunk 27]
```diff
@@ -134,37 +172,117 @@ These permissions are for the cluster and can't be applied granularly. For examp
...
+
+### Cluster monitoring
+
+The monitor cluster permissions apply to read-only operations, such as checking cluster health and getting information about usage on nodes or tasks executing in the cluster.
```
Comment (JeffH-AWS) [inline _security/access-control/permissions.md:282]: Should they be called monitor cluster permissions or cluster monitor permissions? Not sure either way, just asking because the header and subsequent properties use the order "cluster monitor" which makes the first few words of this sentence take a second to digest.

File: _security/access-control/users-roles.md [hunk 28]
```diff
@@ -129,14 +129,17 @@ Role | Description
 `snapshot_management_full_access` | Grants full permissions to all snapshot management actions.
 `snapshot_management_read_access` | Grants permissions to view policies but not create, modify, start, stop, or delete them.
 `point_in_time_full_access` | Grants full permissions to all Point in Time operations.
+`security_analytics_full_access` | Grants full permissions to all Security Analytics functionality.
+`security_analytics_read_access` | Grants permissions to view the various components in Security Analytics, such as detectors, alerts, and findings. It also includes permissions that allow users to search for detectors and rules. This role does not allow a user to perform actions such as modfying or deleting a detector.
+`security_analytics_ack_alerts` | Grants permissions to view and acknowledge alerts.
+
```
Comment (JeffH-AWS) [inline _security/access-control/users-roles.md:135]: This table is also missing pipe characters at the beginning and end of lines. I know you didn't add the table but like the other comment it should have those pipe characters at the beginning and end of each line per https://www.markdownguide.org/extended-syntax/#tables

File: _security/access-control/users-roles.md [hunk 29]
```diff
@@ -129,14 +129,17 @@ Role | Description
...
 
 For more detailed summaries of the permissions for each role, reference their action groups against the descriptions in [Default action groups]({{site.url}}{{site.baseurl}}/security/access-control/default-action-groups/).
 
 
 ## Sample roles
 
-The following examples show how you might set up a read-only and a bulk access role.
-
+The following examples show how you might set up a read-only role and a bulk access role.
```
Comment (JeffH-AWS) [inline _security/access-control/users-roles.md:142]: Maybe "demonstrate" instead of "show?"

**Review by JeffH-AWS — CHANGES_REQUESTED** [review]
Great work Chris! I left a few comments. Also, don't forget to rebase or clean up broken links:

```
LinkChecker: [Warning] Found 11 dead links:
#sample-request, linked in ./_api-reference/snapshots/get-snapshot-status.md
#sample-request, linked in ./_api-reference/snapshots/restore-snapshot.md
#sample-request, linked in ./_api-reference/snapshots/verify-snapshot-repository.md
/upgrade-to/upgrade-to/#upgrade-to-opensearch, linked to in ./_upgrade-to/upgrade-to.md
/upgrade-to/upgrade-to/#upgrade-paths, linked to in ./_upgrade-to/upgrade-to.md
/im-plugin/ism/error-prevention/index/#force_merge/, linked to in ./_dashboards/im-dashboards/forcemerge.md
/security/configuration/tls/#configure-admin-certificates, linked to in ./_security/access-control/api.md, ./_security/configuration/system-indices.md
/im-plugin/index-templates/, linked to in ./_security/access-control/permissions.md
/availability-and-recovery/snapshots/snapshot-restore/#shared-file-system, linked to in ./_api-reference/snapshots/create-repository.md
/availability-and-recovery/snapshots/snapshot-restore/#amazon-s3, linked to in ./_api-reference/snapshots/create-repository.md
/security/configuration/tls#configure-admin-certificates, linked to in ./_troubleshoot/security-admin.md
```

File: _security/access-control/users-roles.md [hunk 30]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `anomaly_full_access` | Grants full permissions to all anomaly detection actions. |
+| `anomaly_read_access` | Grants permissions to view detectors, but not create, modify, or delete detectors. |
+| `all_access` | Grants full access to the cluster, including all cluster-wide operations, permission to write to all cluster indexes, and permission to write to all tenants. For more information on access using the REST API, see [Access control for the API]({{site.url}}{{site.baseurl}}/security/access-control/api/#access-control-for-the-api). |
+| `cross_cluster_replication_follower_full_access` | Grants full access to perform cross-cluster replication actions on the follower cluster. |
+| `cross_cluster_replication_leader_full_access` | Grants full access to perform cross-cluster replication actions on the leader cluster. |
+| `observability_full_access` | Grants full access to perform actions on Observability objects such as visualizations, notebooks, and operational panels. |
+| `observability_read_access` | Grants permission to view Observability objects such as visualizations, notebooks, and operational panels, but not create, modify, or delete them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:115]:
```suggestion
| `observability_read_access` | Grants permission to view Observability objects such as visualizations, notebooks, and operational panels but not to create, modify, or delete them. |
```

File: _security/access-control/users-roles.md [hunk 31]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `opensearch_dashboards_read_only` | A special role that prevents users from making changes to visualizations, dashboards, and other OpenSearch Dashboards objects. See `opensearch_security.readonly_mode.roles` in `opensearch_dashboards.yml`. Pair with the `opensearch_dashboards_user` role. |
+| `opensearch_dashboards_user` | Grants permissions to use OpenSearch Dashboards: cluster-wide searches, index monitoring, and write to various OpenSearch Dashboards indices. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:117]:
```suggestion
| `opensearch_dashboards_user` | Grants permissions to use OpenSearch Dashboards: cluster-wide searches, index monitoring, and write to various OpenSearch Dashboards indexes. |
```

File: _security/access-control/users-roles.md [hunk 32]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `logstash` | Grants permissions for Logstash to interact with the cluster: cluster-wide searches, cluster monitoring, and write to the various Logstash indices. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:118]:
```suggestion
| `logstash` | Grants permissions for Logstash to interact with the cluster: cluster-wide searches, cluster monitoring, and write to the various Logstash indexes. |
```

File: _security/access-control/users-roles.md [hunk 33]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `manage_snapshots` | Grants permissions to manage snapshot repositories, take snapshots, and restore snapshots. |
+| `readall` | Grants permissions for cluster-wide searches like `msearch` and search permissions for all indices. |
+| `readall_and_monitor` | Same as `readall`, but with added cluster monitoring permissions. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:121]:
```suggestion
| `readall_and_monitor` | Same as `readall` but with added cluster monitoring permissions. |
```

File: _security/access-control/users-roles.md [hunk 34]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `security_rest_api_access` | A special role that allows access to the REST API. See `plugins.security.restapi.roles_enabled` in `opensearch.yml` and [Access control for the API]({{site.url}}{{site.baseurl}}/security/access-control/api/#access-control-for-the-api). |
+| `reports_read_access` | Grants permissions to generate on-demand reports, download existing reports, and view report definitions, but not to create report definitions. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:123]:
```suggestion
| `reports_read_access` | Grants permissions to generate on-demand reports, download existing reports, and view report definitions but not to create report definitions. |
```

File: _security/access-control/users-roles.md [hunk 35]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `reports_instances_read_access` | Grants permissions to generate on-demand reports and download existing reports, but not to view or create report definitions. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:124]:
```suggestion
| `reports_instances_read_access` | Grants permissions to generate on-demand reports and download existing reports but not to view or create report definitions. |
```

File: _security/access-control/users-roles.md [hunk 36]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `reports_full_access` | Grants full permissions to reports. |
+| `asynchronous_search_full_access` | Grants full permissions to all asynchronous search actions. |
+| `asynchronous_search_read_access` | Grants permissions to view asynchronous searches but not to submit, modify, or delete async searches. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:127]:
```suggestion
| `asynchronous_search_read_access` | Grants permissions to view asynchronous searches but not to submit, modify, or delete them. |
```

File: _security/access-control/users-roles.md [hunk 37]
```diff
@@ -101,42 +101,45 @@ See [Create role mapping]({{site.url}}{{site.baseurl}}/security/access-control/a
...
+| `index_management_full_access` | Grants full permissions to all index management actions, including Index State Management (ISM), transforms, and rollups. |
+| `snapshot_management_full_access` | Grants full permissions to all snapshot management actions. |
+| `snapshot_management_read_access` | Grants permissions to view policies but not create, modify, start, stop, or delete them. |
```
Comment (natebower) [inline _security/access-control/users-roles.md:130]:
```suggestion
| `snapshot_management_read_access` | Grants permissions to view policies but not to create, modify, start, stop, or delete them. |
```

**Review by natebower — COMMENTED** [review]
@cwillum Please see my comments and changes and let me know if you have any questions. Thanks!

File: _security/access-control/permissions.md [hunk 38]
```diff
@@ -105,10 +123,20 @@ These permissions are for the cluster and can't be applied granularly. For examp
 - cluster:admin/opendistro/alerting/monitor/get
 - cluster:admin/opendistro/alerting/monitor/search
 - cluster:admin/opendistro/alerting/monitor/write
+
+### Asynchronous Search permissions
```
[the remark this answers is not in the document]
Comment (cwillum) [reply _security/access-control/permissions.md:127]: Yes, Asynchronous Search is the plugin name.

**Comment by cwillum** [comment]
@here Just waiting for a final sign-off on this from doc team review comments.

**Comment by cwillum** [comment]
@JeffH-AWS please verify and OK changes so I can get this one out the door. Thanks.

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by cwillum** [comment]
Will revisit merge conflicts after 2.7 release.
